src/query/functions/aggregation/count_values.go

```diff
@@ -88,6 +88,10 @@ type countValuesNode struct {
 	controller *transform.Controller
 }
 
+func (n *countValuesNode) Params() parser.Params {
```

Comment:
Shouldn't this function (and other snowflakes existing outside of various `base.go` files) be refactored to the `ProcessBlock(queryCtx *models.QueryContext, ID parser.NodeID, b block.Block)` model?

Reply:
Yeah it's a bit annoying unfortunately; I guess it's a downside that we have to deal with in return for the nice properties of functions calling their own downstream processing

Reply:
Yeah for a brief moment I was like "we should just explicitly walk the graph instead" and then I realized that would be a large rewrite 😄

Reply:
Yeah, was looking at making a few nicety fixes to DAG generation/execution/mapping, and it's a pretty hefty beast unfortunately to make it more usable for things like this (although it would definitely not go astray!)